build/Common.prod.props

```diff
@@ -16,7 +19,7 @@
     <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
     <Authors>OpenFeature Authors</Authors>
     <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
-    <Version>$(VersionNumber)</Version>
```

Comment:
Are you sure about this change? I'm worried about the impact that might have on the prod release. I'm not sure how this, `VersionNumber`, and `VersionPrefix` work in conjuction.

Reply:
Very fair question! I use this pattern all the time, albeit for internal packages, so just double-checked:

- [`src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultAssemblyInfo.targets`](https://github.com/dotnet/sdk/blob/47245cb134e77f41b5d664413e1fa98b781c8ec1/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultAssemblyInfo.targets)
  ```csproj
  <!--
  ***********************************************************************************************
  Microsoft.NET.DefaultAssemblyInfo.targets
  
  WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
            created a backup copy.  Incorrect changes to this file will make it
            impossible to load or build your projects from the command-line or the IDE.
  
  Copyright (c) .NET Foundation. All rights reserved.
  ***********************************************************************************************
  -->
  <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  
    <PropertyGroup Condition=" '$(Version)' == '' ">
      <VersionPrefix Condition=" '$(VersionPrefix)' == '' ">1.0.0</VersionPrefix>
      <Version Condition=" '$(VersionSuffix)' != '' ">$(VersionPrefix)-$(VersionSuffix)</Version>
      <Version Condition=" '$(Version)' == '' ">$(VersionPrefix)</Version>
    </PropertyGroup>
  
    <PropertyGroup>
      <Authors Condition=" '$(Authors)'=='' ">$(AssemblyName)</Authors>
      <Company Condition=" '$(Company)'=='' ">$(Authors)</Company>
      <AssemblyTitle Condition=" '$(AssemblyTitle)' == '' ">$(AssemblyName)</AssemblyTitle>
      <Product Condition=" '$(Product)' == ''">$(AssemblyName)</Product>
    </PropertyGroup>
  
  </Project>
  ```
- https://learn.microsoft.com/dotnet/core/tools/dotnet-pack
  > - **`--version-suffix <VERSION_SUFFIX>`**
  > 
  >   Defines the value for the `VersionSuffix` MSBuild property. The effect of this property on the package version depends on the values of the `Version` and `VersionPrefix` properties, as shown in the following table:
  > 
  >   | Properties with values              | Package version                     |
  >   |-------------------------------------|-------------------------------------|
  >   | None                                | `1.0.0`                             |
  >   | `Version`                           | `$(Version)`                        |
  >   | `VersionPrefix` only                | `$(VersionPrefix)`                  |
  >   | `VersionSuffix` only                | `1.0.0-$(VersionSuffix)`            |
  >   | `VersionPrefix` and `VersionSuffix` | `$(VersionPrefix)-$(VersionSuffix)` |
  > 
  >   If you want to use `--version-suffix`, specify `VersionPrefix` and not `Version` in the project file. For example, if `VersionPrefix` is `0.1.2` and you pass `--version-suffix rc.1` to `dotnet pack`, the package version will be `0.1.2-rc.1`.
  > 
  >   If `Version` has a value and you pass `--version-suffix` to `dotnet pack`, the value specified for `--version-suffix` is ignored.
  >

Reply:
Sorry, I didn't see your previous comment [here](https://github.com/open-feature/dotnet-sdk/pull/173#issuecomment-1901231142).